lib/chef/platform/provider_mapping.rb

```diff
@@ -263,7 +263,8 @@ def platforms
               :default => {
                 :mount => Chef::Provider::Mount::Solaris,
                 :package => Chef::Provider::Package::SmartOS,
-                :group => Chef::Provider::Group::Usermod
+                :group => Chef::Provider::Group::Usermod,
+                :user => Chef::Provider::User::SmartOS
```

Comment:
Instead of modifying the old provider mapping you should add a providers line to the provider:

https://www.chef.io/blog/2015/02/10/chef-12-provider-resolver/